privacyidea/lib/tokens/tantoken.py

```diff
@@ -126,9 +126,15 @@ def update(self, param, reset_failcount=True):
             # init tokens with tans
             tans = param.get("tans").split()
             tan_dict = {k: v for k, v in enumerate(tans)}
+            # Avoid to generate TANs, since we get the from params
+            param["tantoken_count"] = 0
+            # Determine the otplen from the TANs
+            if len(tans) > 0:
+                param["otplen"] = len(tans[0])
```

Comment:
Hm, should we maybe validate the lengths of all supplied TANs? Or should we handle this as an instance of SISO? :)

Reply:
I like siso a lot.
What should we do, if there were TANs with different length.
These tans are only imported tans. So the import file has to be crafted carefully anyways.

If you come up with a method that verifies tans with the same length, go for it. What should we do then, if we encounter different tan-length (which will neverTM happen) ;-)

Reply:
The first and probably only use case will be the import of a csv file.
To my knowledge the CSV file import does behave nicely if it encounters a ParameterError or any other exception. It will simply break the complete import.

Far from perfect. But this is probably another issue.